h2o-core/src/main/java/water/rapids/ast/prims/filters/dropduplicates/DropDuplicateRowsTask.java

```diff
@@ -0,0 +1,70 @@
+package water.rapids.ast.prims.filters.dropduplicates;
+
+import water.MRTask;
+import water.fvec.Chunk;
+import water.fvec.Frame;
+import water.fvec.NewChunk;
+
+/**
+ * Performs the row de-duplication itself.
+ */
+public class DropDuplicateRowsTask extends MRTask<DropDuplicateRowsTask> {
+  final Frame chunkBoundaries;
+  private final int[] comparedColumnIndices;
+
+  /**
+   * @param chunkBoundaries       Frame with border values of each chunk
+   * @param comparedColumnIndices Columns indices to include in row deduplication comparison
+   */
+  public DropDuplicateRowsTask(final Frame chunkBoundaries, final int[] comparedColumnIndices) {
+    this.chunkBoundaries = chunkBoundaries;
+    this.comparedColumnIndices = comparedColumnIndices;
+  }
+
+  @Override
+  public void map(Chunk[] chunks, NewChunk[] newChunks) {
+    final int chunkLength = chunks[0].len();
+    final int chunkId = chunks[0].cidx();
+    for (int row = 0; row < chunkLength; row++) {
+      final boolean equal;
+      if (chunkId == 0 && row == 0) {
+        for (int columnIndex = 0; columnIndex < chunks.length; columnIndex++) {
+          chunks[columnIndex].extractRows(newChunks[columnIndex], row);
+        }
+        continue;
+      } else if (chunkId != 0 && row == 0) {
+        equal = compareFirstRowWithPreviousChunk(chunks, row, chunkId);
+      } else {
+        equal = compareRows(chunks, row);
+      }
+
+      if (!equal) {
+        for (int columnIndex = 0; columnIndex < chunks.length; columnIndex++) {
+          chunks[columnIndex].extractRows(newChunks[columnIndex], row);
+        }
+      }
+
+    }
+  }
+
+  private boolean compareFirstRowWithPreviousChunk(final Chunk[] chunks, final int row, final int chunkId) {
+    for (int column : comparedColumnIndices) {
+      final double previousValue = chunkBoundaries.vec(column).chunkForChunkIdx(chunkId - 1).atd(0);
+      final double curentValue = chunks[column].atd(row);
```

Comment:
does this work for string columns as well? don't see any tests for String

Reply:
We don't support string columns, unfortunately. For this, we need Merge.Sort to support String columns. And it doesn't. Sorting will fail. For now, I propose adding a check and a nice error message would do. WDYT ?

This brings me to another problem - All chunks have all the methods, and some of them will throw `IllegalArgumentException`  (I know, Java has UnsupportedOperationException for this, don't ask me why this is there). Here, I am using `atd`, thus for chunks of `long`, not all 64-bit signed integers can be represented as double. We have to somehow check the type of the chunk and act accordingly.

Reply:
yes, if we dont support some column type we should definitely check at the beginning and throw a reasonable error

Reply:
I've added the check and a test for it. Also added BAD columns (obvious) and UUID columns (can't be sorted as well).

Reply:
Also, I've added handling for T_CAT and T_TIME separately. In case of T_CAT, we could handle it as double (2^31 of ints fits into IEEE 754 double), but long is closer and a bit cheaper to convert I'd say, therefore used at8 for time..